### PR TITLE
make Firefox client work again with new state structure

### PR DESCRIPTION
**Pull request by jlongster**

### CODE REVIEW

**Comment by jasonLaster** [comment]
Looking good. 

+1 extracting loaded frames

File: public/js/components/Sources.js [hunk 1]
```diff
@@ -62,9 +62,9 @@ let File = (props) => {
     )
   );
 };
-File = React.createFactory(File);
```
Comment (jlongster) [inline public/js/components/Sources.js]: linter was complaining that `File` is a builtin type that we are overriding so I had to change these. Will probably pull them out later today too

Comment (jasonLaster) [inline public/js/components/Sources.js]: sounds good. good thing we didn't have a Date and Blob icon :)

**Comment by jlongster** [comment]
Cool! Seems like we'll get a good feeling about all of this pretty quickly.

**Pull request merged**